src/protocol/sort/compose.rs

```diff
@@ -44,24 +46,24 @@ impl<'a, F: Field> Compose<'a, F> {
     pub async fn execute(
         &mut self,
         ctx: ProtocolContext<'_, Replicated<F>, F>,
-    ) -> Result<(), BoxError> {
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
         let mut random_permutations =
             get_two_of_three_random_permutations(self.rho.len(), &ctx.prss());
         let mut random_permutations_copy = random_permutations.clone();
 
-        Shuffle::new(self.sigma)
+        self.sigma = Shuffle::new(take(&mut self.sigma))
```

Comment:
the compiler is usually smart enough to figure out if you do
```suggestion
        self.sigma = Shuffle::new(self.sigma)
```

but do you really need to assign it back to `self`? seems like it might be more clear if you do
```suggestion
        let sigma = Shuffle::new(self.sigma)
```
last thing: it's also an option to explicitly extract the values from self:
```rust
let Self { sigma, rho } = self;

Reply:
Ended up removing these variables from members to send as function parameters